virttest/utils_test/__init__.py

```diff
@@ -1947,149 +1956,147 @@ def _tarball_download(self, url, destination):
         self.base_name = tarball.members[0].name.split("/")[0]
         tarball.close()
 
-    @error_context.context_aware
     def download_stress(self):
         """
         Download stress tool
         """
         url = self.download_url
         tmp_path = data_dir.get_tmp_dir()
-        error_context.context('Download stress tool from %s' % url,
-                              logging.info)
+        logging.info('Download stress tool from %s', url)
         download_method = getattr(self, "_%s_download" % self.download_type)
         download_method(url, tmp_path)
-        error_context.context('Copy stress tool to guest', logging.info)
-        self.vm.wait_for_login()
-        self.vm.copy_files_to(os.path.join(tmp_path, self.base_name), self.dst_path)
+        source = os.path.join(tmp_path, self.base_name)
+        if self.remote_host:
+            logging.info('Copy stress tool to remote host')
+
+            args = (self.remote_host.__getitem__('server_ip'), 'scp',
+                    self.remote_host.__getitem__('server_user'),
+                    self.remote_host.__getitem__('server_pwd'), "22",
+                    source, self.dst_path)
+            self.copy_files_to(args)
+        else:
+            logging.info('Copy stress tool to work dir of host/guest')
+            self.copy_files_to(source, self.dst_path)
```

Comment:
@balamuruhans, i'm confused here. Generally speaking, when we copy files to remote host or guest, we use "scp", but when we copy files to host, we just use "cp". So i don't think copy to host/guest the same logic.
 virt_vm.copy_files_to()  can as a reference.

Reply:
Hi @aliang123 right, I have defined self.copy_files_to() based on host//VM/remote host as below,
By default in Stress(), for local host 
```python
self.copy_files_to = shutil.copytree
```
For VM, the VMStress() would override the `self.copy_files_to`
```python
self.copy_files_to = self.vm.copy_files_to
```
For remote host, HostStress() would override 
```python
self.copy_files_to = remote.copy_files_to
```
So, each would take care of copying the files to host/VM/remote host respectively.